### PR TITLE
adding a-star try one

### PR DESCRIPTION
**Pull request by elkyhalberstam**

### CODE REVIEW

File: src/main/java/smartsnake/astar/halberstam/AstarBrain.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            Node right = new Node(new Point(current.getLocation().x + 1, current.getLocation().y), food);
+            Node left = new Node(new Point(current.getLocation().x - 1, current.getLocation().y), food);
+
+            neighbors.add(up);
+            neighbors.add(down);
+            neighbors.add(right);
+            neighbors.add(left);
+
+            for (int i = 0; i < neighbors.size(); i++) {
+                if (closeNodes.contains(neighbors.get(i))) {
```
Comment (andrewoid) [inline src/main/java/smartsnake/astar/halberstam/AstarBrain.java:54]: Call neighbors.get(i) only once

File: src/test/java/smartsnake/astar/halberstam/AstarBrainTest.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.verify;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class AstarBrainTest {
+    //given
+        Snake snake = new Snake();
+        Food food = new Food(12, 5);
+        Garden garden = new Garden(100, 200);
```
Comment (andrewoid) [inline src/test/java/smartsnake/astar/halberstam/AstarBrainTest.java:16]: These should be mocks

**Comment by elkyhalberstam** [comment]
I ended up working on the a-star on my neural network branch

**Comment by andrewoid** [comment]
I'm not approving this because it fails a test. 
If this is fixed on your other branch then we will deal with it once your other branch is merged.

**Comment by elkyhalberstam** [comment]
This is fixed on the other branch and I merged my other branch into main.
What should I do?

**Pull request closed without merging**